rust/otap-dataflow/crates/engine/src/node.rs

```diff
@@ -30,13 +36,38 @@ pub trait Node {
     async fn send_control_msg(&self, msg: NodeControlMsg) -> Result<(), SendError<NodeControlMsg>>;
 }
 
+/// NodeId consists of NodeId and NodeIndex integer.
+#[derive(Clone, Debug)]
+pub struct NodeId {
+    /// A unique integer.
+    pub(crate) index: NodeIndex,
+
+    /// A unique name as defined by otap_df_config.
+    pub name: NodeName,
+}
+
+/// Index in the NodeDefs vector
+#[derive(Copy, Clone, Debug, Eq, Ord, PartialEq, PartialOrd, Hash)]
+pub struct NodeIndex(u16);
```

Comment:
@jmacd Why use `u16` instead of `usize` if this is going to be used as an index for the vec?